internal/staging/stdaddr/addressv0.go

```diff
@@ -308,6 +313,132 @@ func (addr *AddressPubKeyEd25519V0) String() string {
 	return addr.Address()
 }
 
+// AddressPubKeySchnorrSecp256k1V0 specifies an address that represents a
+// payment destination which imposes an encumbrance that requires a valid
+// EC-Schnorr-DCRv0 signature for a specific secp256k1 public key.
+//
+// This is commonly referred to as pay-to-pubkey-schnorr-secp256k1.
+type AddressPubKeySchnorrSecp256k1V0 struct {
+	pubKeyID         [2]byte
+	pubKeyHashID     [2]byte
+	serializedPubKey []byte
+}
+
+// Ensure AddressPubKeySchnorrSecp256k1V0 implements the Address interface.
+var _ Address = (*AddressPubKeySchnorrSecp256k1V0)(nil)
+
+// NewAddressPubKeySchnorrSecp256k1V0Raw returns an address that represents a
+// payment destination which imposes an encumbrance that requires a valid
+// EC-Schnorr-DCRv0 signature for a specific secp256k1 public key using version
+// 0 scripts.
+//
+// The provided public key MUST be a valid secp256k1 public key serialized in
+// the _compressed_ format or an error will be returned.
+//
+// See NewAddressPubKeySchnorrSecp256k1V0 for a variant that accepts the public
+// key as a concrete type instance instead.
+//
+// This function can be useful to callers who already need the serialized public
+// key for other purposes to avoid the need to serialize it multiple times.
+func NewAddressPubKeySchnorrSecp256k1V0Raw(serializedPubKey []byte,
+	params AddressParamsV0) (*AddressPubKeySchnorrSecp256k1V0, error) {
+
+	// Attempt to parse the provided public key to ensure it is both a valid
+	// serialization and that it is a valid point on the secp256k1 curve.
+	pubKey, err := secp256k1.ParsePubKey(serializedPubKey)
+	if err != nil {
+		str := fmt.Sprintf("failed to parse public key: %v", err)
+		return nil, makeError(ErrInvalidPubKey, str)
+	}
+
+	// Ensure the provided serialized public key is in the compressed format.
+	// This probably should be returned from secp256k1, but do it here to avoid
+	// API churn.  The pubkey is known to be valid since it parsed above, so
+	// it's safe to simply examine the leading byte to get the format.
+	//
+	// Notice that both the uncompressed and hybrid forms are intentionally not
+	// supported.
+	switch serializedPubKey[0] {
+	case secp256k1.PubKeyFormatCompressedEven:
+	case secp256k1.PubKeyFormatCompressedOdd:
+	default:
+		str := fmt.Sprintf("serialized public key %x is not a valid format",
+			serializedPubKey)
+		return nil, makeError(ErrInvalidPubKeyFormat, str)
+	}
+
+	return &AddressPubKeySchnorrSecp256k1V0{
+		pubKeyID:         params.AddrIDPubKeyV0(),
+		pubKeyHashID:     params.AddrIDPubKeyHashSchnorrV0(),
+		serializedPubKey: pubKey.SerializeCompressed(),
```

Comment:
I notice you're re-serializing here but didn't in the Ecdsa variant. Is this intended?

Reply:
It is indeed intentional.  Hybrid/Uncompressed public keys are not allowed by consensus for the Schnorr secp256k1 signature checking opcode unlike the ECDSA secp256k1 one, so this forces it to the only supported serialization format.

Reply:
Looking at this again, I see you're right.  I read this as being in `NewAddressPubKeySchnorrSecp256k1V0` for some reason where it is intentionally doing what I explained in the previous comment.  This isn't any reason to serialize it again here.  Updated.